src/librustc/ty/context.rs

```diff
@@ -1112,6 +1101,20 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
         let cname = self.crate_name(LOCAL_CRATE).as_str();
         self.sess.consider_optimizing(&cname, msg)
     }
+
+    pub fn lang_items(self) -> Rc<middle::lang_items::LanguageItems> {
+        // Right now we insert a `with_ignore` node in the dep graph here to
+        // ignore the fact that `get_lang_items` below depends on the entire
+        // crate.  For now this'll prevent false positives of recompiling too
+        // much when anything changes.
+        //
+        // Once red/green incremental compilation lands we should be able to
+        // remove this because while the crate changes often the lint level map
+        // will change rarely.
+        self.dep_graph.with_ignore(|| {
```

Comment:
I wanted to make sure I called this out as well. This `with_ignore` was necessary to not instantly break all of the incremental tests!

I imagine though that with red/green this'll always generate the same value, so the node's always green even though it needs to be recomputed?